src/Components/Blazor/Blazor/src/Http/WebAssemblyHttpMessageHandler.cs

```diff
@@ -113,10 +106,18 @@ private string[][] GetHeadersAsStringArray(HttpRequestMessage request)
             }
         }
 
-        private static byte[] AllocateArray(string length)
-        {
-            return new byte[int.Parse(length)];
-        }
+        /// <remarks>
+        /// While it may be tempting to remove this method because it appears to be unused,
+        /// this method is referenced by client code and must persist.
+        /// </remarks>
+#pragma warning disable IDE0051 // Remove unused private members
+        private static byte[] AllocateArray(string length) => new byte[int.Parse(length)];
+#pragma warning restore IDE0051 // Remove unused private members
+
+        private string[][] GetHeadersAsStringArray(HttpRequestMessage request)
+            => (from header in request.Headers.Concat(request.Content?.Headers ?? Enumerable.Empty<KeyValuePair<string, IEnumerable<string>>>())
+                from headerValue in header.Value // There can be more than one value for each name
+                select new[] { header.Key, headerValue }).ToArray();
```

Comment:
eww linq

Reply:
That was me again. Sometimes that syntax just feels right. I mean, usually it doesn't, but *sometimes*...